File: backend/app/models/feed.py
# backend/app/models/feed.py
from sqlalchemy import String, DateTime, Integer
from sqlalchemy.orm import Mapped, mapped_column, relationship
from typing import Optional
from app.extensions import db
from app.models.base import Base
from sqlalchemy.orm import relationship

class Feed(Base):
    __tablename__ = "feed"

    id: Mapped[int] = mapped_column(primary_key=True)
    feed_flag_status_id: Mapped[int] = mapped_column(db.ForeignKey("feed_flag_status.id"))
    url: Mapped[str] = mapped_column(String(2083), unique=True)
    last_parsed_file_hash: Mapped[Optional[str]] = mapped_column(String(32))
    parsing_priority: Mapped[int] = mapped_column(Integer, default=0)
    container_id: Mapped[Optional[str]] = mapped_column(String(12))
    created_at: Mapped[DateTime] = mapped_column(DateTime, server_default=db.func.now())
    updated_at: Mapped[DateTime] = mapped_column(DateTime, server_default=db.func.now(), onupdate=db.func.now())
    is_parsing: Mapped[Optional[bool]] = mapped_column(db.Boolean)

    flag_status = relationship("FeedFlagStatus", back_populates="feeds")
    channels = relationship("Channel", back_populates="feed")
    logs = relationship(
        "FeedLog",
        back_populates="feed",
        cascade="all, delete-orphan",
        order_by="desc(FeedLog.finished_at)"
    )

    @property
    def recent_logs(self):
        return self.logs[:2]  # adjust count as needed

class FeedFlagStatus(Base):
    __tablename__ = "feed_flag_status"

    id: Mapped[int] = mapped_column(primary_key=True)
    created_at: Mapped[DateTime] = mapped_column(DateTime, server_default=db.func.now())
    updated_at: Mapped[DateTime] = mapped_column(DateTime, server_default=db.func.now(), onupdate=db.func.now())
    status: Mapped[str] = mapped_column(String(50))

    feeds = relationship("Feed", back_populates="flag_status")

class FeedLog(Base):
    __tablename__ = "feed_log"

    id: Mapped[int] = mapped_column(primary_key=True)
    feed_id: Mapped[int] = mapped_column(db.ForeignKey("feed.id"))
    http_status: Mapped[Optional[int]] = mapped_column(db.Integer) # HTTP status based on result of parse
    is_success: Mapped[Optional[bool]] = mapped_column(db.Boolean) # T or F based on the Reparse Result
    parse_errors: Mapped[Optional[int]] = mapped_column(db.Integer, default=0)
    parse_error_message: Mapped[Optional[str]] = mapped_column(String(255)) # new field to hold message
    started_at: Mapped[Optional[DateTime]] = mapped_column(DateTime)
    finished_at: Mapped[Optional[DateTime]] = mapped_column(DateTime)
    parsed_by: Mapped[Optional[str]] = mapped_column(String(255)) # This will come as an Auth0 ID

    feed = relationship("Feed", back_populates="logs")

File: frontend/types/feed.ts
export interface FeedLog {
  id: number;
  feed_id: number;
  http_status?: number;
  is_success?: boolean;
  parse_errors?: number;
  parse_error_message?: string;
  started_at?: string;
  finished_at?: string;
  parsed_by?: string;
  last_finished_parse_time?: string;
}

export interface Feed {
  id: number;
  flag_status: string;
  url: string;
  last_parsed_file_hash?: string;
  parsing_priority?: number;
  container_id?: string;
  created_at?: string;
  updated_at?: string;
  is_parsing?: boolean;
  logs?: FeedLog[];
  recent_logs?: FeedLog[];
}

export interface FeedFlagStatus {
  id: number;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface FeedFilters {
  status: string;
  parsing_priority?: number;
  is_parsing?: boolean;
  sort_by?: string;
  sort_order?: string;
  searchTerm?: string;
  limit?: number;
  page?: number;
}

export interface FeedStatusInfo {
  id: number;
  label: string;
  className: string;
}

export const FEED_STATUS_MAP: Record<string, FeedStatusInfo> = {
  active:           { id: 1, label: "Active", className: "bg-green-500 text-white" },
  "always-parse":   { id: 2, label: "Always Parse", className: "bg-green-500 text-white" },
  spam:             { id: 3, label: "Spam", className: "bg-yellow-400 text-white" },
  "pending-archive":{ id: 4, label: "Pending Archive", className: "bg-yellow-400 text-blue-900" },
  archived:         { id: 5, label: "Archived", className: "bg-gray-500 text-white" },
  takedown:         { id: 6, label: "Takedown", className: "bg-yellow-400 text-white" },
  parse_error:      { id: 7, label: "Parse Error", className: "bg-red-400 text-black" },
  fetch_error:      { id: 8, label: "Fetch Error", className: "bg-red-400 text-white" },
  "":               { id: 0, label: "Unknown", className: "bg-gray-300 text-black" }
};